### PR TITLE
CPM-829: As an administrator, I want only specific users to access the identifier generator features

### PR DESCRIPTION
**Pull request by mmetayer**
Feature branch for ACls

- [x] CPM-831 #18741 
- [x] CPM-832 #18861 
- [x] CPM-838 #18788 
    To fix after rebase:
    - [x] in structure tab, disable drag'n'drop of properties in readoly mode
    - [x] in conditions tab, make operator and value fields readonly for family, status and simple select conditions
- [x] CPM-930 #18978
- [x] CPM-897 #19063 
- [x] CPM-961: Add ACL checks on backend endpoint (plus test)

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [x] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

TODO Handle readonly:
- [x] Family Property #18995
    - [x] components/identifier-generator/front/src/feature/tabs/structure/edit/FamilyPropertyEdit.tsx
- [x] TextTransformation component

### CODE REVIEW

File: components/identifier-generator/back/src/Infrastructure/Controller/DeleteIdentifierGeneratorController.php [hunk 1]
```diff
@@ -5,11 +5,15 @@
...
 use Akeneo\Pim\Automation\IdentifierGenerator\Domain\Repository\IdentifierGeneratorRepository;
+use Akeneo\Platform\Bundle\FrameworkBundle\Security\SecurityFacadeInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+;
```
Comment (Beninho) [inline components/identifier-generator/back/src/Infrastructure/Controller/DeleteIdentifierGeneratorController.php:16]: remove the semicolon

**Review by Beninho — APPROVED** [review]

File: components/identifier-generator/back/src/Infrastructure/Symfony/Resources/config/form_extensions.yml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-    # Drop this line to display menu
-    aclResourceId: this_is_here_to_hide_feature
-    position: 100
+    parent: pim-menu-automation-settings-block
+    aclResourceId: "pim_identifier_generator_view || pim_identifier_generator_manage"
+    position: 20
     config:
       title: pim_title.akeneo_identifier_generator_index
       to: akeneo_identifier_generator_index
+      new: true
```
Comment (pierallard) [inline components/identifier-generator/back/src/Infrastructure/Symfony/Resources/config/form_extensions.yml:10]: Thanks :heart:

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/config-provider.js [hunk 3]
```diff
@@ -10,7 +10,13 @@ define(['jquery', 'pim/security-context', 'pim/feature-flags'], function ($, Sec
...
   const filterByGranted = extensions => {
     return extensions.filter(extension => {
-      return null === extension.aclResourceId || SecurityContext.isGranted(extension.aclResourceId);
+      if (null === extension.aclResourceId) {
+        return true;
+      }
+
+      return extension.aclResourceId
+        .split(/\|\|/)
+        .reduce((currentValue, resourceId) => currentValue || SecurityContext.isGranted(resourceId.trim()), false);
```
Comment (pierallard) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/config-provider.js:19]:
```suggestion
        .split(/ *\|\| */)
        .reduce((currentValue, resourceId) => currentValue || SecurityContext.isGranted(resourceId), false);
```

**Review by pierallard — APPROVED** [review]

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 4]
```diff
@@ -127,30 +136,49 @@ const ListPage: React.FC<ListPageProps> = ({onCreate}) => {
                     </Table.Cell>
                     <Table.ActionCell>
                       <Button onClick={goToEditPage(code)} ghost>
-                        {translate('pim_common.edit')}
-                      </Button>
-                      <Button onClick={onDelete(code)} ghost level="danger">
-                        {translate('pim_common.delete')}
+                        {translate(isManageIdentifierGeneratorAclGranted ? 'pim_common.edit' : 'pim_common.view')}
```
[the remark this answers is not in the document]
Comment (mmetayer) [reply components/identifier-generator/front/src/feature/pages/ListPage.tsx:139]: these translation keys are defined in the `Akeneo\Connectivity\Connection` namespace, I'd rather not use them

**Comment by mmetayer** [comment]
green build: https://app.circleci.com/pipelines/github/akeneo/pim-community-dev/48793/workflows/6eb2b7ee-dcda-4609-8a9f-90331fce51b6

**Pull request merged**